### PR TITLE
Live:104 - converted star rating icon to svg

### PR DESCRIPTION
**Pull request by dskamiotis**
## Why are you doing this?

Currently Star Rating is using icons which cause issues when rendering in certain browser and ios/android versions, with strange fallback icons when not loading (square box).

Issues regarding Icons and how SVg are more desirable:

Icon Accessibility: do not support screen readers, SVG's, however are accessible to a screen reader (they are treated as images)
Font icons are not as scalable as they are font sized as opposed to vector based.
In terms of performance icons can increase HTTP server requests.for caching
Icons are vulnerable to anti-aliasing techniques causing blurring
In addition svg's allow for a high degree of modifications without triggering additional server requests so can have a performance advantage

We could provide PNG icons as fallback, but since we have SVG's as part of source not necessary.

## Changes

- Customised styles for the new SVG element
- Included SvgStar instead of the icon as a tag

## Screenshots

| Before | After |
| --- | --- |
|  |  |

### CODE REVIEW

**Comment by dskamiotis** [comment]
@benwuersching I have added you as we re factored the construction of the SvgStar for star rating and tried to keep as similar to DCR and code live.

**Comment by benwuersching** [comment]
@dskamiotis This looks good, thank you. The new stars seem ever so slightly further apart. Could they move slightly closer to each other?

**Comment by JamieB-gu** [comment]
> Could they move slightly closer to each other?

I think this is actually because the component in Source contains padding on the left and right. @SiAdcock do you think we should adjust this here in CSS or tweak the SVG in Source?

File: src/components/starRating.tsx [hunk 1]
```diff
@@ -1,56 +1,53 @@
...
-
-    &:nth-of-type(1) {
-        padding-left: ${remSpace[1]};
-    }
+    line-height: 0;
 
-    &:nth-of-type(5) {
-        padding-right: ${remSpace[1]};
+    svg {
+        height: 2rem;
```
Comment (webb04) [inline src/components/starRating.tsx:20]: Do we lose some of that right padding?

File: src/components/starRating.tsx [hunk 2]
```diff
@@ -16,7 +16,12 @@ const starStyles = css`
     display: inline-block;
     line-height: 0;
 
+    &:nth-of-type(5) {
+        padding-right: 2px;
+    }
+;
     svg {
+        margin: 0 -2px;
```
Comment (JamieB-gu) [inline src/components/starRating.tsx:24]: You may want to convert this to `rem` to handle scaling.

File: src/components/starRating.tsx [hunk 3]
```diff
@@ -1,56 +1,58 @@
...
+    line-height: 0;
 
     &:nth-of-type(5) {
-        padding-right: ${remSpace[1]};
+        padding-right: 2px;
+    }
+;
+    svg {
+        margin: 0 -0.125rem;
+        height: 2rem;
```
Comment (webb04) [inline src/components/starRating.tsx:25]: This brings the size a bit closer to dotcom and the current apps-rendering articles:

```suggestion
        height: 1.75rem;
```

**Review by webb04 — APPROVED** [review]

**Pull request merged**